snapshot-file-test/src/main/java/com/hazelcast/jet/tests/snapshot/file/GeneratedFilesVerifier.java

```diff
@@ -122,6 +130,21 @@ private void verifyQueue() {
         }
     }
 
+    private void archiveLoadedFile(Path path) throws IOException {
+        // add suffix to not override files which can have the same name (it is mostly for dynamic cluster)
+        String filename = path.toFile().getName() + "_" + UuidUtil.newUnsecureUuidString().substring(0, 8);
```

Comment:
instead of creating a full uuid we can just add a int counter which resets at every new sub-directory creation.